In [1]:
import pandas as pd

In [2]:
titanic_df = pd.read_csv("train.csv")

In [3]:
titanic_df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

### basic information

In [4]:
titanic_df.shape

(891, 12)

In [5]:
titanic_df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [6]:
titanic_df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [7]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## specific informaton

In [8]:
titanic_df[["Age","Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [9]:
titanic_df[["Age","Fare"]].describe()

Age        Fare
count  714.000000  891.000000
mean    29.699118   32.204208
std     14.526497   49.693429
min      0.420000    0.000000
25%     20.125000    7.910400
50%     28.000000   14.454200
75%     38.000000   31.000000
max     80.000000  512.329200

**description

1. average age of passengers is about 30,which suggest most of the passenger are young between 20 to 40

2. average fare of passengers is about 32 ,where the most ticket price range are from around 17 to 80 dollar 

In [10]:
titanic_df.value_counts("Survived")

Survived
0    549
1    342
Name: count, dtype: int64

In [11]:

titanic_df.value_counts(["Sex","Survived"],normalize=True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

death rate of male population is double the survival rate of female 

In [12]:
titanic_df[titanic_df["Sex"] == "female"].value_counts("Survived")

Survived
1    233
0     81
Name: count, dtype: int64

In [13]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts("Survived")

Survived
0    427
1    217
Name: count, dtype: int64

out of 644 passenger boarding from south-hampton 427 people died while 217 people survived

In [14]:
titanic_df.value_counts(["Embarked","Survived"],normalize=True)

Embarked  Survived
S         0           0.480315
          1           0.244094
C         1           0.104612
          0           0.084364
Q         0           0.052868
          1           0.033746
Name: proportion, dtype: float64

In [15]:
titanic_df[titanic_df["Embarked"] == "S"]["Survived"].value_counts(normalize=True)


Survived
0    0.663043
1    0.336957
Name: proportion, dtype: float64

In [16]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts("Sex")

Sex
male      441
female    203
Name: count, dtype: int64

In [19]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts("Survived")

Survived
0    427
1    217
Name: count, dtype: int64

### nunique() , unique()

In [20]:
titanic_df["Embarked"].unique()

array(['S', 'C', 'Q', nan], dtype=object)

In [21]:
titanic_df["Embarked"].nunique()

3

In [22]:
titanic_df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [23]:
titanic_df.notna().sum()

PassengerId    891
Survived       891
Pclass         891
Name           891
Sex            891
Age            714
SibSp          891
Parch          891
Ticket         891
Fare           891
Cabin          204
Embarked       889
dtype: int64

In [31]:
group_by_gender = titanic_df[["Sex","Fare","Age","Survived"]].groupby(by = "Sex")

In [32]:
group_by_gender.mean()

Fare        Age  Survived
Sex                                   
female  44.479818  27.915709  0.742038
male    25.523893  30.726645  0.188908

In [33]:
grooup_by_gender.value_counts(["Survived"])

Sex     Survived
female  1           233
        0            81
male    0           468
        1           109
Name: count, dtype: int64

In [46]:
group_by_embarked = titanic_df[["Embarked","Fare","Age","Survived"]].groupby(by = ["Embarked","Survived"])

In [49]:
group_by_embarked.mean()

Fare        Age
Embarked Survived                      
C        0         35.443335  33.666667
         1         79.720926  28.973671
Q        0         13.335904  30.325000
         1         13.182227  22.500000
S        0         20.743987  30.203966
         1         39.547081  28.113184

In [51]:
group_by_embarked.mean(numeric_only = True)

Fare        Age
Embarked Survived                      
C        0         35.443335  33.666667
         1         79.720926  28.973671
Q        0         13.335904  30.325000
         1         13.182227  22.500000
S        0         20.743987  30.203966
         1         39.547081  28.113184

### correlation and covariance

In [52]:
titanic_df[["Age","Fare"]].cov()

Age         Fare
Age   211.019125    73.849030
Fare   73.849030  2469.436846

In [53]:
titanic_df[["Age","Fare"]].corr()

Age      Fare
Age   1.000000  0.096067
Fare  0.096067  1.000000

In [55]:
sample = titanic_df.sample(n=55,random_state=99)


In [56]:
sample[["Age","Fare"]].corr()

Age      Fare
Age   1.000000  0.134789
Fare  0.134789  1.000000

### crosstab


In [57]:
pd.crosstab(
    titanic_df.Pclass,
    titanic_df.Survived,
    margins=True
)

Survived    0    1  All
Pclass                 
1          80  136  216
2          97   87  184
3         372  119  491
All       549  342  891